graphql/src/execution/execution.rs

```diff
@@ -201,7 +202,8 @@ lazy_static! {
 
     // Cache query results for recent blocks by network.
     // The `VecDeque` works as a ring buffer with a capacity of `QUERY_CACHE_BLOCKS`.
-    static ref QUERY_BLOCK_CACHE: RwLock<QueryBlockCache> = RwLock::new(QueryBlockCache(vec![]));
+    static ref QUERY_BLOCK_CACHE: TimedRwLock<QueryBlockCache> =
+                                    TimedRwLock::new("query_block_cache", QueryBlockCache(vec![]));
     static ref QUERY_HERD_CACHE: QueryCache<Arc<QueryResult>> = QueryCache::new();
     static ref QUERY_LFU_CACHE: Mutex<LfuCache<QueryHash, WeightedResult>> = Mutex::new(LfuCache::new());
```

Comment:
Should we do something similar for this `Mutex`? We see quite a few historical queries, and we might actually be contending on this lock.

And similar, can we be sure that the herd cache isn't heavily contended, too?

With logging for additional caches, make sure it's easy to search for the log messages for lock contention and to connect them to the cache we are contending on.

Reply:
Instrumented the mutexes, I tested the log messages and they are searchable and informative.